test/e2e/upgrade_management_cluster_test.go

```diff
@@ -2,340 +2,137 @@ package e2e
+			E2EConfig:                 e2eConfig,
+			ClusterctlConfigPath:      clusterctlConfigPath,
+			BootstrapClusterProxy:     bootstrapClusterProxy,
+			ArtifactFolder:            artifactFolder,
+			SkipCleanup:               true, // parameterize after merging https://github.com/kubernetes-sigs/cluster-api/pull/7373
+			InitWithProvidersContract: "v1alpha4",
+			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.4.7/clusterctl-{OS}-{ARCH}",
+			PreInit:                   getPreInitFunc(ctx),
+			PreWaitForCluster:         getPreWaitForCluster(ctx),
+			PreUpgrade:                getPreUpgrade(ctx),
+			MgmtFlavor:                osType,
+			WorkloadFlavor:            "ubuntu",
+		}
 	})
-	var (
-		ctx                 = context.TODO()
-		specName            = "metal3"
-		namespace           = "metal3"
-		clusterName         = "test1"
-		workloadClusterName = "test2"
-		clusterctlLogFolder string
-		kubernetesVersion   = e2eConfig.GetVariable(KubernetesVersion)
-	)
-	It("Should create a management cluster and then upgrade capi and capm3", func() {
-		/*---------------------------------------------*
-		| Create a target cluster with v1a4 clusterctl |
-		*----------------------------------------------*/
-		Logf("Starting v1a5 to v1b1 upgrade tests")
-
-		clusterctlBinaryURLTemplate := e2eConfig.GetVariable("INIT_WITH_BINARY")
-		clusterctlBinaryURLReplacer := strings.NewReplacer("{OS}", runtime.GOOS, "{ARCH}", runtime.GOARCH)
-		clusterctlBinaryURL := clusterctlBinaryURLReplacer.Replace(clusterctlBinaryURLTemplate)
-		clusterctlBinaryPath := "/tmp/clusterctltmp"
-
-		Logf("Downloading clusterctl binary from %s", clusterctlBinaryURL)
-		err := downloadFile(clusterctlBinaryPath, clusterctlBinaryURL)
-		Expect(err).ToNot(HaveOccurred(), "failed to download temporary file")
-		defer os.Remove(clusterctlBinaryPath) // clean up
-
-		err = os.Chmod(clusterctlBinaryPath, 0744)
-		Expect(err).ToNot(HaveOccurred(), "failed to chmod temporary file")
-
-		By("Creating a cluster")
-		Logf("Getting the cluster template yaml")
-		upgradeClusterTemplate := clusterctl.ConfigClusterWithBinary(ctx, clusterctlBinaryPath, clusterctl.ConfigClusterInput{
-			KubeconfigPath:       bootstrapClusterProxy.GetKubeconfigPath(),
-			ClusterctlConfigPath: clusterctlConfigPath,
-			// define template variables
-			Namespace:                namespace,
-			ClusterName:              clusterName,
-			KubernetesVersion:        kubernetesVersion,
-			Flavor:                   "ubuntu",
-			ControlPlaneMachineCount: pointer.Int64Ptr(1),
-			WorkerMachineCount:       pointer.Int64Ptr(1),
-			InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
-			// setup clusterctl logs folder
-			LogFolder: clusterctlLogFolder,
-		})
+})
 
-		Expect(upgradeClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
+// Install the split-yaml tool to split our bmhosts_crs.yaml file.
+func installSplitYaml() {
```

Comment:
with respect to Go naming conventions:
```suggestion
func installSplitYAML() {
```